### ISSUE TITLE
💅 Biome 2.0 with JSX h factory that comes NOT from React

### ISSUE THREAD

**Issue by Jagget** [biomejs/biome#6438]
### Environment information

```bash
Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v22.15.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/11.4.1"
```

### Rule name

lint/correctness/noUnusedImports

### Playground link

no

### Expected result

After update to v2 I got a warning that I didn't see in v1.9:

The error:
```sh
src/util/DOM/SsrDataScript.tsx:1:43 lint/correctness/noUnusedImports  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ⚠ Several of these imports are unused.
  
  > 1 │ import { Build, type FunctionalComponent, h } from '@stencil/core';
      │                                           ^
    2 │ 
    3 │ const getScriptText = (dataId: number, data: unknown) => {
  
  ℹ Unused imports might be the result of an incomplete refactoring.
  
  ℹ Unsafe fix: Remove the unused imports.
  
    1 │ import·{·Build,·type·FunctionalComponent,·h·}·from·'@stencil/core';
      │                                           --     
```

With my tsconfig:
```json
{  
  "compilerOptions": {
    "jsx": "react",  
    "jsxFactory": "h"
  }
}
```

and I tried 2 options for JSX factory, got the same warning.

biome.json
```json
{  
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",  
  "assist": {  
    "actions": {  
      "source": {  
        "organizeImports": "on"  
      }  
    }  
  },
  "javascript": {  
    "jsxRuntime": "transparent/reactClassic",  // tried both options, with the same result
    "formatter": {  
      "quoteStyle": "single"  
    }  
  }
}
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by scriptcoded** [comment]
Can confirm I'm seeing the exact same issue. I would expect Biome to respect the `compilerOptions.jsxFactory` setting in tsconfig.json.
